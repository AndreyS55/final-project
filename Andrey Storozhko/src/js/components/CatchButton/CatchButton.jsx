import React from 'react';

class CatchButton extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isDisable: false
    };
  }

  catchPokemon = (e) => {
    e.stopPropagation();
    this.setState({ isDisable: !this.state.isDisable }, this.props.catchPokemon({
      name: this.props.name,
      date: new Date().toLocaleDateString(),
      pokemonId: this.props.id,
      id: this.props.id
    }));
  };

  render() {
    return (
      <button
        onClick={this.catchPokemon}
        disabled={this.state.isDisable || this.props.date || this.props.catched.length}
      >
        Catch!
      </button>
    );
  }
}

export default CatchButton;
