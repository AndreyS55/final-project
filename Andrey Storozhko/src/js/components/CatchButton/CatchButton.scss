.catchButton {
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 70%;
  padding: 0 5px;
  border: 1px solid #ffba80;
  background-color: rgba(255, 215, 182, 0.5);

  &:disabled {
    &:hover {
      background-color: rgba(255, 215, 182, 0.5);
      cursor: default;
    }

    &:active {
      position: relative;
      top: 0;
    }
  }

  &:hover {
    background-color: rgba(255, 215, 182, 0.9);
    cursor: pointer;
  }

  &:active {
    position: relative;
    top: 1px;
  }
}

.catchButtonIcon {
  width: 30px;
}