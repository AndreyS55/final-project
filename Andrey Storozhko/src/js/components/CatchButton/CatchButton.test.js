import React from 'react';
import CatchButton from './CatchButton';

it('should render a CatchButton', () => {
  const component = shallow(<CatchButton />);
  expect(component).toMatchSnapshot();
});
