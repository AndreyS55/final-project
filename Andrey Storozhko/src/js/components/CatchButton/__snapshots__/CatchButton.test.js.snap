// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a CatchButton 1`] = `
<button
  className="catchButton"
  disabled={1}
  onClick={[Function]}
>
  <div>
    Caught
  </div>
  <img
    alt="Catch button icon"
    className="catchButtonIcon"
    src="test-file-stub"
  />
</button>
`;
